deps: add simdutf dependency (#45803)
nodejs/node: tools/dep_updaters/README.md

   example).

[`c61870c`]: https://github.com/nodejs/node/commit/c61870c376e2f5b0dbaa939972c46745e21cdbdd

> ## simdutf

Reviewer: I think there should be a README in the deps/simdutf directory as well that points to this document and/or the script.

Author: Added a readme. Thanks for the review.

The `update-simdutf.sh` script takes the target version to update as its only
argument, downloads it from the [GitHub repo](https://github.com/simdutf/simdutf)

> and uses it to replace the contents of `deps/uv/`. The contents are replaced

Reviewer 2: ```suggestion
and uses it to replace the contents of `deps/simdutf/`. The contents are replaced
```

entirely except for the `*.gyp` and `*.gypi` build files, which are part of the
Node.js build definitions and are not present in the upstream repo.

> For example, in order to update to version `1.44.2`, the following command can

Reviewer 2: ```suggestion
For example, in order to update to version `2.0.6`, the following command can
```

be run:

```bash
./tools/dep_updaters/update-simdutf.sh 2.0.6
```

Once the script has run (either manually, or by CI in which case a PR will have
been created with the changes), do the following:

1. Check the [changelog](https://github.com/simdutf/simdutf/releases/tag/v2.0.3) for
   things that might require changes in Node.js.

> 2. If necessary, update `common.gypi` and `simdutf.gyp` with build-related changes.

Reviewer 2: ```suggestion
2. If necessary, update `simdutf.gyp` with build-related changes.
```
